scripts/create-cloudgov-env.sh

```diff
@@ -0,0 +1,139 @@
+# infrastructure needed to run weahter.gov. It can serve for documentation for running
+# NOTE: This script was written for MacOS and to be run at the root directory. 
+
+# A function to generate a random string, used for cron key, password, etc.
+generate_string()
+{
+  if [ -z "$1" ] ; then
+    if command -v uuidgen >/dev/null ; then
+      NEW_STRING=$(uuidgen)
+      export NEW_STRING
+    else
+      echo "cannot find uuidgen utility:  You will need to generate some random strings and put them in the CRON_KEY, HASH_SALT, and ROOT_USER_PASS environment variables, then re-run this script."
+      exit 1
+    fi
+  fi
+}
+
+if [ -z "$1" ]; then
+    echo 'Please specify a new space to create (i.e. lmm)' >&2
+    exit 1
+fi
+
+if [ ! $(command -v gh) ] || [ ! $(command -v jq) ] || [ ! $(command -v cf) ]; then
+    echo "jq, cf, and gh packages must be installed. Please install via your preferred manager."
+    exit 1
+fi
+
+upcase_name=$(printf "%s" "$1" | tr '[:lower:]' '[:upper:]')
+
+read -p "Are you on a new branch? We will have to commit this work. (y/n) " -n 1 -r
+echo
+if [[ ! $REPLY =~ ^[Yy]$ ]]
+then
+    git checkout -b new-dev-sandbox-$1
+fi
+
+cf target -o nws-weathergov
+
+read -p "Are you logged in to the nws-weathergov CF org above? (y/n) " -n 1 -r
+echo
+if [[ ! $REPLY =~ ^[Yy]$ ]]
+then
+    cf login -a https://api.fr.cloud.gov --sso
+fi
+
+gh auth status
+read -p "Are you logged into a Github account with access to weather-gov/weather.gov? (y/n) " -n 1 -r
+echo
+if [[ ! $REPLY =~ ^[Yy]$ ]]
+then
+    gh auth login
+fi
+
+echo "Creating manifest for $1..."
+cp manifests/manifest.template.yaml manifests/manifest-$1.yaml
+sed -i '' "s/ENVIRONMENT/$1/" "manifests/manifest-$1.yaml"
+
+echo "Creating new cloud.gov space for $1..."
+cf create-space $1
+cf target -o "nws-weathergov" -s $1
+cf bind-security-group public_networks_egress nws-weathergov --space $1
+cf bind-security-group trusted_local_networks_egress nws-weathergov --space $1
+
+echo "Creating new cloud.gov DB for $1. This usually takes about 5 minutes..."
+cf create-service aws-rds small-mysql database
+
+until cf service database | grep -q 'The service instance status is succeeded'
+do
+  echo "Database not up yet, waiting..."
+  sleep 30
+done
+
+echo "Creating storage for file uploads in $1..." 
+cf create-service s3 basic-sandbox storage
+
+echo "Creating new cloud.gov credentials for $1..."
+generate_string "$CRON_KEY"
+CRON_KEY=${CRON_KEY:-$NEW_STRING}
+generate_string "$HASH_SALT"
+HASH_SALT=${HASH_SALT:-$NEW_STRING}
+generate_string "$ROOT_USER_PASS"
+ROOT_USER_PASS=${ROOT_USER_PASS:-$NEW_STRING}
+ROOT_USER_NAME=${ROOT_USER_NAME:-root}
+cf create-user-provided-service secrets -p "{\"CRON_KEY\": \"$CRON_KEY\", \"HASH_SALT\": \"$HASH_SALT\", \"ROOT_USER_NAME\": \"$ROOT_USER_NAME\", \"ROOT_USER_PASS\": \"$ROOT_USER_PASS\"}"
+
+echo "Database create succeeded and credentials created. Deploying the get.gov application to the new space $1..."
```

Comment:
[non-blocking question] weather.gov instead of get.gov?

Reply:
Yep :) still stuck in previous project headspace I guess!